consensus/XDPoS/utils/pool.go

```diff
@@ -22,7 +24,17 @@ func NewPool() *Pool {
 	}
 }
 func (p *Pool) Get() map[string]map[common.Hash]PoolObj {
-	return p.objList
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+	dataCopy := maps.Clone(p.objList)
+	for k, v := range dataCopy {
+		dataCopy[k] = make(map[common.Hash]PoolObj, len(dataCopy[k]))
+		for k2, v2 := range v {
+			dataCopy[k][k2] = v2.DeepCopy().(PoolObj)			
+		}
+	}
+
```

Comment:
`maps.Clone` copies keys and values, this not needed, since we will copy them manually at once:

```go
	dataCopy := make(map[string]map[common.Hash]PoolObj, len(p.objList))
	for k1, v1 := range p.objList {
		dataCopy[k1] = make(map[common.Hash]PoolObj, len(v1))
		for k2, v2 := range v1 {
			dataCopy[k1][k2] = v2.DeepCopy().(PoolObj)
		}
	}
```